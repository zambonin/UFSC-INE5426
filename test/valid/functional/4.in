int a, b, c, d
a = 2
b = 4
lambda int x, y -> x + y
c = λ(a, b)
λ()
lambda int x, y -> x - y
d = λ(a, b)
