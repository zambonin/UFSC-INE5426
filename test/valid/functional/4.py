exec(open('../../../src/scope_manager.py', 'r').read())




a = 2
b = 4
def 位(x, y):
  return (x + y)

c = 位(a, b)
def 位(x, y):
  return (x - y)

d = 位(a, b)
