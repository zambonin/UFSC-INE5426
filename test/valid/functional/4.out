int var: a, b, c, d
= a 2
= b 4
int fun: lambda (params: int x, int y)
  ret + x y
= c lambda[2 params] a b
int fun: lambda (params: int x, int y)
  ret - x y
= d lambda[2 params] a b
